src/main/java/org/prebid/server/bidder/bidmachine/BidmachineBidder.java

```diff
@@ -0,0 +1,227 @@
+                        .body(body)
+                        .headers(resolveHeaders())
+                        .payload(outgoingRequest)
+                        .build());
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        return Result.of(httpRequests, errors);
+    }
+
+    private static void validateImp(Imp imp) {
+        final Banner banner = imp.getBanner();
+        if (banner == null) {
+            return;
+        }
+
+        if (banner.getW() == null && banner.getH() == null) {
+            final List<Format> format = banner.getFormat();
+            if (format == null) {
+                throw new PreBidException("Impression with id: " + imp.getId()
+                        + " has following error: Banner width and height is not provided and"
+                        + " banner format is missing. At least one is required");
+            }
+            if (format.isEmpty()) {
+                throw new PreBidException("Impression with id: " + imp.getId() + " has following error:"
+                        + " Banner width and height is not provided and banner format array is empty. "
+                        + "At least one is required");
+
+            }
+        }
+    }
+
+    private BidRequest createRequest(Imp imp, ExtImpPrebid extPrebid, BidRequest request) {
+        return request.toBuilder()
+                .imp(Collections.singletonList(modifyImp(imp, extPrebid)))
+                .build();
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpPrebid extPrebid) {
+        if (extPrebid != null && Objects.equals(extPrebid.getIsRewardedInventory(), 1)) {
+            final Banner banner = imp.getBanner();
+            final List<Integer> bannerBattr = banner == null ? null : banner.getBattr();
+            final List<Integer> resolvedBannerBattr = resolveBattrList(bannerBattr);
+
+            final Video video = imp.getVideo();
+            final List<Integer> videoBattr = video == null ? null : video.getBattr();
+            final List<Integer> resolvedVideoBattr = resolveBattrList(videoBattr);
+
+            if (resolvedBannerBattr != null || resolvedVideoBattr != null) {
+                final Imp.ImpBuilder impBuilder = imp.toBuilder();
+                if (resolvedBannerBattr != null) {
+                    impBuilder.banner(banner.toBuilder().battr(resolvedBannerBattr).build());
+                }
+                if (resolvedVideoBattr != null) {
+                    impBuilder.video(video.toBuilder().battr(resolvedVideoBattr).build());
+                }
+
+                return impBuilder.build();
+            }
+        }
+
+        return imp;
+    }
+
+    private static List<Integer> resolveBattrList(List<Integer> battr) {
+        if (battr != null && !hasRewardedBattr(battr)) {
+            final List<Integer> updatedBattr = new ArrayList<>(battr);
```

Comment:
If battr is null, we should create collection and add 16, but in our case `new ArrayList<>(battr)` will throw NPE for this case